### PR TITLE
PCHR-2879: Fix styling for Leave Request Modal

### PR DESCRIPTION
**Pull request by igorpavlov**
## Overview

This PR fixes the styling for Leave Request Modal. Additionally it handles the case when the working day cannot be retrieved after date selection (ex. when the date is selected outside of any contact's job contract).

Here is the list of related PRs: 
https://github.com/civicrm/civihr/pull/2286
https://github.com/compucorp/civihr-employee-portal-theme/pull/274
https://github.com/compucorp/org.civicrm.shoreditch/pull/67

## Before

b1

Same issues appear for SSP.

### Styling

**Admin**

a1

**SSP**

a2

### Error handling

a3

### Styling

```css
<div class="col-xs-6 col-sm-4 chr_leave-request-date-picker-container">
<!-- Class "col-xs-6" removed in all applicable places -->
<div class="col-sm-4 chr_leave-request-date-picker-container">
```

Other styling changes were made in https://github.com/compucorp/org.civicrm.shoreditch/pull/67 after the directive inherited **crm_custom-select** selector class as per https://github.com/civicrm/civihr/pull/2286

## Comments

This PR also handles the exception when a working day cannot be retrieved.

This PR also fixes the issue with edit and view modes. Dates are now returned with time, so their string values are not equal anymore even if it is the same day, so time has to be omitted:

```js
// was
vm.request.from_date === vm.request.to_date
// now
_convertDateToServerFormat(vm.request.from_date) === _convertDateToServerFormat(vm.request.to_date)
```

Also, the time values were updated in edit mode on **load**, which should never happen.

------

✅ Jasmine Tests - amended and passed
✅ JS distributive files - included
✅ Manual Tests - passed
✅ CSS distributive files - included
⏹ Backstop Tests - omitted
⏹ PHP Unit Tests - not needed

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js [hunk 1]
```diff
@@ -564,6 +564,10 @@ define([
 
       timeObject.disabled = true;
       timeObject.time = '';
+      timeObject.min = '0';
+      timeObject.max = '0';
+      timeObject.maxAmount = '0';
+      timeObject.amount = '0';
```
Comment (bmnepali) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js]: Can we keep these assignments in alphabetical order?

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js [hunk 2]
```diff
@@ -572,7 +576,7 @@ define([
         timeObject.amount = timeObject.maxAmount || '0';
```
Comment (bmnepali) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js]: Do we relally need `timeObject.amount = timeObject.maxAmount **|| '0'**;` ? As for me if 
 `response.number_of_hours.toString()` does not exist `timeObject.maxAmount` will already be "'0'" right?

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-popup-details-tab.component.spec.js [hunk 3]
```diff
@@ -532,6 +532,38 @@ define([
...
+                  });
+
+                  it('flushes and disables time and deduction fields', function () {
+                    expect(timeToObject.disabled).toBeTruthy();
+                    expect(timeToObject.time).toBe('');
+                    expect(timeToObject.min).toBe('0');
+                    expect(timeToObject.max).toBe('0');
+                    expect(timeToObject.maxAmount).toBe('0');
+                    expect(timeToObject.amount).toBe('0');
+                  });
```
Comment (bmnepali) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-popup-details-tab.component.spec.js]: Remove the duplicate test case.
```js
 it('flushes and disables time and deduction fields', function () {
   ...
 })
```

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js [hunk 4]
```diff
@@ -563,20 +565,28 @@ define([
...
-        timeObject.time = (type === 'to' ? timeObject.max : timeObject.min);
-      }).catch(handleError);
+
+        if (!skipTimeValuesUpdate) {
+          timeObject.time = (type === 'to' ? timeObject.max : timeObject.min);
+          timeObject.amount = timeObject.maxAmount || '0';
+        }
+      }).catch(handleError).finally(function () {
+        skipTimeValuesUpdate = false;
+      });
```
Comment (bmnepali) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js]:
```js
})
.catch(handleError)
.finally(function () {
   skipTimeValuesUpdate = false;
 });
```

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js [hunk 5]
```diff
@@ -618,7 +628,10 @@ define([
      * Sets day selection mode: multiple days or a single day
      */
     function setDaySelectionMode () {
-      if ((vm.isMode('edit') && vm.request.from_date === vm.request.to_date) ||
+      if (
+        ((vm.isMode('edit') || vm.isMode('view')) &&
+        _convertDateToServerFormat(vm.request.from_date) ===
+        _convertDateToServerFormat(vm.request.to_date)) ||
         (vm.isMode('create') && vm.isLeaveType('sickness'))) {
```
Comment (bmnepali) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js]: Here the the condition has been complex. Can we separate them into separate functions?

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-popup-details-tab.component.spec.js (hunk 3 above)
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-popup-details-tab.component.spec.js]: Have no idea how it happened though...

**Pull request merged**